io/eolearn/tests/test_sentinelhub_process.py

```diff
@@ -164,6 +167,47 @@ def test_upsampling_downsampling(self, resampling_type: ResamplingType, stats: L
         width, height = self.size
         assert bands.shape == (4, height, width, 1)
 
+    @pytest.mark.parametrize(
+        ["geometry", "stats"],
+        [
+            (
+                Geometry(
+                    BBox([easting + 50 * 10, northing, easting + size[0] * 10, northing + 50 * 10], crs=crs).geometry,
+                    crs=crs,
+                ),
+                [0.0, 0.1547, 0.0],
+            ),
+            (
+                Geometry(
+                    BBox(
+                        [easting - 20, northing - 20, easting + (size[0] + 10) * 10, northing + (size[1] + 10) * 10],
+                        crs=crs,
+                    ).geometry,
+                    crs=crs,
+                ),
+                [0.0836, 0.1547, 0.0794],
+            ),
```

Comment:
If i understand correctly, this test requests the same area but with two different geometries?

Reply:
yes, the first geometry is smaller than the bounding box, while the second is larger than the bounding box.

We wanted to test that in the second case you get the same results as requesting only the bounding box.